From 817f39e49599c3308b5d619163a2426269633067 Mon Sep 17 00:00:00 2001
From: Kevin Benton <kevin@benton.pub>
Date: Tue, 30 May 2017 21:38:45 -0700
Subject: [PATCH] Provide fallback for disabled port security extension

The push notification logic always assumed the port security object
would exist but it is not present on the port when the extension is
disabled. This defaults it to true like the server side code.[1]

1.
https://github.com/openstack/neutron/blob/c430e9b8d41c139284e840be37629afcdbc96b37/neutron/plugins/ml2/rpc.py#L142

Change-Id: Ice89ad9dd486ad5fcac534ef5f7d8aae3b6b0f97
Closes-Bug: #1694420
---
 neutron/agent/rpc.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/neutron/agent/rpc.py b/neutron/agent/rpc.py
index bcfc6c3e1df..d0810885339 100644
--- a/neutron/agent/rpc.py
+++ b/neutron/agent/rpc.py
@@ -254,7 +254,8 @@ def get_device_details(self, context, device, agent_id, host=None):
             'allowed_address_pairs': [{'mac_address': o.mac_address,
                                        'ip_address': o.ip_address}
                                       for o in port_obj.allowed_address_pairs],
-            'port_security_enabled': port_obj.security.port_security_enabled,
+            'port_security_enabled': getattr(port_obj.security,
+                                             'port_security_enabled', True),
             'qos_policy_id': port_obj.qos_policy_id,
             'network_qos_policy_id': net_qos_policy_id,
             'profile': port_obj.binding.profile,
